tests/unit_tests/constraint/test_fcialg.py

```diff
@@ -559,3 +674,82 @@ def test_fci_complex(self):
+        Based on Figure 6 from :footcite:`Zhang2008`
+
+        """
+
+        import pywhy_graphs
+
+        logging.getLogger().setLevel(logging.DEBUG)
+
+        # Pretend this is a MAG - refactor if MAGs are developed
+        G = ADMG()
+        G.add_edge("A", "C", G.directed_edge_name)
+        G.add_edge("A", "B", G.bidirected_edge_name)
+        G.add_edge("B", "D", G.directed_edge_name)
+        G.add_edge("C", "D", G.directed_edge_name)
+        assert pywhy_graphs.networkx.m_separated(G, {"A"}, {"D"}, {"B", "C"})
+
+        sample = dummy_sample(G)
+        context = make_context().variables(data=sample).build()
+        oracle = Oracle(G)
+        ci_estimator = oracle
+        fci = FCI(ci_estimator=ci_estimator, max_iter=np.inf, selection_bias=False)
+        fci.fit(sample, context)
+        pag = fci.graph_
+
+        expected_G = PAG()
+        expected_G.add_edge("A", "B", expected_G.circle_edge_name)
+        expected_G.add_edge("B", "A", expected_G.circle_edge_name)
+        expected_G.add_edge("C", "A", expected_G.circle_edge_name)
+        expected_G.add_edge("A", "C", expected_G.circle_edge_name)
+        expected_G.add_edge("B", "D", expected_G.directed_edge_name)
+        expected_G.add_edge("C", "D", expected_G.directed_edge_name)
+
+        assert pag.nodes() == expected_G.nodes()
+        assert pag.edges() == expected_G.edges()
+
+    def test_fci_selection_bias(self):
+        """
+        Based on Figure 1 from :footcite:`Zhang2008`, with extra edge R -> D
+
+        The DAG (over observed and selected variables) is
+        A -> Ef <-> R -> D, Ef -> Sel, where Sel is a selection variable.
+
+        The MAG is A - Ef -> R -> D.
+
+        The PAG is A o-o Ef o-o R o-o D.
+
+        References
+        ----------
+        .. footbibliography::
+
+        """
+
+        # Pretend we have a MAG - refactor if in future MAGs are implemented
+        G = PAG()
+        G.add_edge("A", "Ef", G.undirected_edge_name)
+        G.add_edge("Ef", "R", G.directed_edge_name)
+        G.add_edge("R", "D", G.directed_edge_name)
+        G._edge_graphs.pop("circle")
+
+        sample = dummy_sample(G)
+        context = make_context().variables(data=sample).build()
+        oracle = Oracle(G)
+        ci_estimator = oracle
+        fci = FCI(ci_estimator=ci_estimator, max_iter=np.inf, selection_bias=True)
+        fci.fit(sample, context)
+        pag = fci.graph_
+
+        expected_pag = PAG()
+        expected_pag.add_edge("A", "Ef", expected_pag.circle_edge_name)
+        expected_pag.add_edge("Ef", "A", expected_pag.circle_edge_name)
+        expected_pag.add_edge("Ef", "R", expected_pag.circle_edge_name)
+        expected_pag.add_edge("R", "Ef", expected_pag.circle_edge_name)
+        expected_pag.add_edge("R", "D", expected_pag.circle_edge_name)
+        expected_pag.add_edge("D", "R", expected_pag.circle_edge_name)
+        assert pag.nodes() == expected_pag.nodes()
+        assert pag.edges() == expected_pag.edges()
```

Comment:
So the expected PAG only has circle edges? Is there an example of a graph, where we know the outcome PAG has "tails" that we can test against?

Reply:
I had another example that I ended up moving into the test for rule 9, but lemme see if i can think of some more complicated examples. If you have any nice examples please let me know.

I will try and implement:
- Fig 2. of https://arxiv.org/pdf/1104.5617.pdf

Reply:
Fig 2 in that paper doesn't have selection bias I think, but tbh I have never seen a PAG example w/ selection bias lol. If it's too hard to think of an example, or find one, then I am okay w/o it. Or we can just take the graph for that PAG in Fig 2 and arbitrarily add a relevant set of nodes that will induce R5, and/or 6(?)

Reply:
Yeah maybe I will just do that for Fig. 2. 

We should probably address this in dodiscover documentation! How are people supposed to use this stuff if there are no good examples anywhere on the internet??

Reply:
Most definitely! I think our repo is starting to come together, where the next steps will be coming up w/ real (albeit simulations most likely) graphical models that reflect how FCI works.

Might be a good mental exercise to "work backwards a valid PAG to the DAG" lolol.